src/core_plugins/kibana/public/dashboard/index.js

```diff
@@ -29,34 +28,75 @@ const app = uiModules.get('app/dashboard', [
 ]);
 
 uiRoutes
-.defaults(/dashboard/, {
-  requireDefaultIndex: true
-})
-.when('/dashboard', {
-  template: indexTemplate,
-  resolve: {
-    dash: function (savedDashboards, config) {
-      return savedDashboards.get();
+  .defaults(/dashboard/, {
+    requireDefaultIndex: true
+  })
+  .when('/dashboard', {
+    template: function ($location) {
+      if ($location.new === true) {
+        return indexTemplate;
+      }
+
+      return false;
+    },
+    resolve: {
+      dash: function (savedDashboards, config, kbnUrl, Notifier, $location, $route) {
+        let defaultDashboard = config.get('dashboard:defaultDashboard', '');
+
+        let notify = new Notifier({
+          location: 'Dashboard'
+        });
+
+        if ($location.search().new === true) {
+          return savedDashboards.get();
+        }
+
+        function forceNew() {
+          $location.search('new', true);
+          $route.reload();
+        }
+
+        if (defaultDashboard !== '' && typeof defaultDashboard !== 'undefined') {
+          return savedDashboards.get(defaultDashboard)
+            .then(function (result) {
+              let dashboardUrl = savedDashboards.urlFor(result.id).substring(1);
+              kbnUrl.change(dashboardUrl);
+            })
+            .catch(function (error) {
+              notify.error(error);
+
+              forceNew();
+            });
+        }
+
+        forceNew();
+      }
     }
-  }
-})
-.when('/dashboard/:id', {
-  template: indexTemplate,
-  resolve: {
-    dash: function (savedDashboards, Notifier, $route, $location, courier) {
-      return savedDashboards.get($route.current.params.id)
-      .catch(courier.redirectWhenMissing({
-        'dashboard' : '/dashboard'
-      }));
+  })
+  .when('/dashboard/:id', {
+    template: indexTemplate,
+    resolve: {
+      dash: function (savedDashboards, Notifier, $route, $location, courier, kbnUrl) {
+        if ($location.search().new === true) {
+          kbnUrl.change('/dashboard');
+          $location.search('new', true);
+        }
+
+        return savedDashboards.get($route.current.params.id)
+          .catch(courier.redirectWhenMissing({
+            'dashboard': '/dashboard'
```

Comment:
No reason for quotes on this property.  I'm actually surprised the linter allowed this.